### PR TITLE
Add GlobalMaxPooling3D and GlobalAveragePooling3D

### LINKED ISSUE #3942: GlobalPooling for 3D inputs
Hello,

I was wondering why there is [GlobalMaxPooling2D](https://keras.io/layers/pooling/#globalmaxpooling2d) and [GlobalAveragePooling2D](https://keras.io/layers/pooling/#globalaveragepooling2d), but no 3D versions of both. 

Looking at the code, one could easily extend both to work with 3D inputs. Should I start a pull request - or am I missing something here?

### PR DESCRIPTION
**Pull request by AnishShah**
Fixes #3942

### CODE REVIEW

**Comment by fchollet** [comment]
Please add tests for these layers in `tests/keras/layers/pooling`.

**Comment by AnishShah** [comment]
I was going to add the tests but couldn't find `test_pooling.py` in `tests/keras/layers`.
Should I create a new file?

**Comment by AnishShah** [comment]
Ok! I just checked `test_convolutional.py` file and it has tests related to pooling.

**Comment by AnishShah** [comment]
All tests passed!

File: keras/layers/pooling.py [hunk 1]
```diff
@@ -519,3 +519,83 @@ def call(self, x, mask=None):
...
+    # Output shape
+        2D tensor with shape:
+        `(nb_samples, channels)`
+    '''
+
+    def call(self, x, mask=None):
+        if self.dim_ordering == 'tf':
+            return K.max(x, axis=[1, 2, 3])
+        else:
+            return K.mean(x, axis=[2, 3, 4])
```
Comment (fchollet) [inline keras/layers/pooling.py]: This should be max.

File: tests/keras/layers/test_convolutional.py [hunk 2]
```diff
@@ -270,6 +270,22 @@ def test_globalpooling_2d():
...
+               input_shape=(3, 4, 11, 12, 10))
+    layer_test(pooling.GlobalMaxPooling3D,
+               kwargs={'dim_ordering': 'tf'},
+               input_shape=(3, 4, 11, 12, 10))
+    layer_test(pooling.GlobalAveragePooling3D,
+               kwargs={'dim_ordering': 'th'},
+               input_shape=(3, 4, 11, 12, 10))
+    layer_test(pooling.GlobalAveragePooling3D,
+               kwargs={'dim_ordering': 'tf'},
+               input_shape=(3, 4, 11, 12, 10))
```
Comment (fchollet) [inline tests/keras/layers/test_convolutional.py]: Use much smaller dimensions in order to speed up the tests (3-4, single channel).

**Comment by fchollet** [comment]
LGTM

**Pull request merged**